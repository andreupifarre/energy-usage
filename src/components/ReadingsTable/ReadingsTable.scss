.ReadingsTable {
  width: 380px;
}
