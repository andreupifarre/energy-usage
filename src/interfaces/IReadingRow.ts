export interface IReadingRow {
  readingDate: string;
  cumulative: number;
  unit: string;
}
