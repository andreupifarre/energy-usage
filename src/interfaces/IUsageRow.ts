export interface IUsageRow {
  date: string;
  energyUsage: number;
}
