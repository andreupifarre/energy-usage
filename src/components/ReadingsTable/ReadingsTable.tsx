import * as React from 'react';
import ReadingRow from '../ReadingRow/ReadingRow';
import { IReadingRow } from '../../interfaces/IReadingRow';
import * as styles from './ReadingsTable.scss';

export interface ReadingsTableProps {
  readings: IReadingRow[];
}

const ReadingsTable: React.SFC<ReadingsTableProps> = ({ readings }) => (
  <table className={styles.ReadingsTable}>
    <tbody>
      <tr>
        <th align="left">Date</th>
        <th align="left">Reading</th>
        <th align="left">Unit</th>
      </tr>
      {readings.map(reading => (
        <ReadingRow
          key={reading.readingDate}
          readingDate={reading.readingDate}
          cumulative={reading.cumulative}
          unit={reading.unit}
        />
      ))}
    </tbody>
  </table>
);

export default ReadingsTable;
