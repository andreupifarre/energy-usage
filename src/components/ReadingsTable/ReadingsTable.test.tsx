import * as React from 'react';
import { shallow } from 'enzyme';
import ReadingsTable from './ReadingsTable';
import { IReadingRow } from '../../interfaces/IReadingRow';

const render = readings => shallow(<ReadingsTable readings={readings} />);
const readings: IReadingRow[] = [
  {
    cumulative: 17580,
    readingDate: '2017-03-28T00:00:00.000Z',
    unit: 'kWh'
  },
  {
    cumulative: 17759,
    readingDate: '2017-04-15T00:00:00.000Z',
    unit: 'kWh'
  },
  {
    cumulative: 18002,
    readingDate: '2017-05-08T00:00:00.000Z',
    unit: 'kWh'
  }
];

describe('ReadingsTable', () => {
  it('should match number of props object pass with number of ReadingRow rendered', () => {
    const elements = render(readings).find('ReadingRow');

    expect(elements.length).toEqual(readings.length);
  });

  it('should render the correct set of prperties for each ReadingRow', () => {
    const rows = render(readings).find('ReadingRow');
    const newRows: IReadingRow[] = rows.map(row => row.props());

    expect(newRows).toEqual(readings);
  });

  it('should render correctly', () => {
    expect(render(readings).getElements()).toMatchSnapshot();
  });
});
