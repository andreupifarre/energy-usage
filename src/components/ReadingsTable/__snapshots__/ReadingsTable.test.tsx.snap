// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingsTable should render correctly 1`] = `
Array [
  <table
    className="ReadingsTable"
  >
    <tbody>
      <tr>
        <th
          align="left"
        >
          Date
        </th>
        <th
          align="left"
        >
          Reading
        </th>
        <th
          align="left"
        >
          Unit
        </th>
      </tr>
      <ReadingRow
        cumulative={17580}
        readingDate="2017-03-28T00:00:00.000Z"
        unit="kWh"
      />
      <ReadingRow
        cumulative={17759}
        readingDate="2017-04-15T00:00:00.000Z"
        unit="kWh"
      />
      <ReadingRow
        cumulative={18002}
        readingDate="2017-05-08T00:00:00.000Z"
        unit="kWh"
      />
    </tbody>
  </table>,
]
`;
